ios/browser/api/ipfs/ipfs_api.h

```diff
@@ -16,8 +16,9 @@ OBJC_EXPORT
 @required
 
 @property(nonatomic, nullable) NSURL* nftIpfsGateway;
+@property(nonatomic, nullable) NSURL* ipfsGateway;
 
-- (nullable NSURL*)resolveGatewayUrlFor:(NSURL*)input;
+- (nullable NSURL*)resolveGatewayUrlFor:(NSURL*)input isForNFT:(BOOL)isForNFT;
```

Comment:
API design here is okay but not great, I dont really expect to have other things here I suppose other than NFT so not terrible. 

Alternatives:
- making an enum with default/nfp types, passing in that instead
- 2 methods: `resolveGatewayUrlFor` and `resolveGatewayUrlForNFT` but then you have to remember to switch them in iOS side
- passing in the gateway directly instead of a bool, but annoying to call
- NSURL extension so that you can just do `nftIpfsGateway.resolve(url)` instead but not all NSURLs and IPFSURL's so not great. Maybe if we made a IPFSURL wrapper type or something 🤷‍♂️

cc @StephenHeaps for opinions

Reply:
Either make an enum or 2 methods I think 👍

Reply:
Updated with two methods in [9b364ec](https://github.com/brave/brave-core/pull/17536/commits/9b364ecf8ce7c97880edea5512c13ecb4f8011bd)